lib/msf/core/exploit/laravel_crypto_killer.rb

```diff
@@ -0,0 +1,188 @@
+    laravel_encrypt(Base64.strict_encode64(session_json_to_encrypt), key, cipher_mode)
+  end
+
+  # Parses Laravel cipher data
+  # @param [String] <laravel_cipher> The base64-encoded Laravel cipher data
+  #
+  # @return [String] The laravel parsed cipher data in JSON format or nil if unsuccessful
+  def parse_laravel_cipher(laravel_cipher)
+    laravel_cipher = CGI.unescape(laravel_cipher) # Decoding URL encoded string
+    begin
+      data = JSON.parse(Base64.strict_decode64(laravel_cipher))
+    rescue JSON::ParserError
+      vprint_error('The JSON inside your base64 is malformed')
+      return
+    rescue StandardError
+      vprint_error('Your base64 laravel_cipher value is malformed')
+      return
+    end
+
+    data['value'] = Base64.strict_decode64(data['value'])
+    data['iv'] = Base64.strict_decode64(data['iv'])
+    data
+  end
+
+  # Parse Laravel APP_KEY value
+  # @param [String] <key> The Laravel APP_KEY
+  #
+  # @return [String] The Laravel parsed APP_KEY
+  def retrieve_key(key)
+    if key.start_with?('base64:')
+      Base64.strict_decode64(key.split(':')[1])
+    elsif key.length == 44
+      Base64.strict_decode64(key)
+    else
+      key.encode('utf-8')
+    end
+  end
+
+  # Decrypts a Laravel ciphered string
+  # @param [String] <laravel_cipher> The Laravel cipher to be decrypted
+  # @param [String] <key> The key used for decryption
+  # @param [String] <cipher_mode> Cipher_mode used for encryption (AES-256-CBC)
+  #
+  # @return [String] The decrypted Laravel cipher or nil if unsuccessful
+  def laravel_decrypt(laravel_cipher, key, cipher_mode)
+    data = parse_laravel_cipher(laravel_cipher)
+    key = retrieve_key(key)
+
+    begin
+      result = aes_decrypt(data['value'], data['iv'], key, cipher_mode)
+      return result
+    rescue StandardError
+      vprint_error('Your key is probably malformed or incorrect.')
+    end
+  end
+
+  # Uses an opened file containing a key on each line to perform a brute-force attack on a given value
+  # @param [String] <value> The encrypted Laravel value
+  # @param [String] <key_file> The file with Laravel APP_KEYs per line used for brute-force decryption
+  # @param [String] <key> The key used for decryption
+  # @param [String] <cipher_mode> Cipher_mode used for encryption (AES-256-CBC)
+  #
+  # @return [String] The valid key if it was identified with the value: {"key":<key>, "value":<value>}
+  def laravel_bruteforce_from_file(value, key_file, cipher_mode)
+    result = nil
+    if File.file?(key_file) # Check if it's a file
+      file_to_test = File.open(key_file, 'r') # Open the file for reading
+      file_to_test.each do |line|
+        key = line.strip
+        decrypted_value = laravel_decrypt(value, key, cipher_mode).force_encoding('utf-8')
+        if decrypted_value
+          result = { 'key' => key, 'value' => decrypted_value }
+          break
+        end
+      rescue StandardError
+        next
+      end
+    end
+    result
+  end
```

Comment:
```suggestion
  def laravel_bruteforce_from_file(value, key_file, cipher_mode)
    if !File.file?(key_file)
      return nil
    end
    
    File.foreach(key_file) do |line|
      key = line.strip
      decrypted_value = laravel_decrypt(value, key, cipher_mode).force_encoding('utf-8')      
      if decrypted_value
        return { 'key' => key, 'value' => decrypted_value }
      end
    rescue StandardError
      next
    end
    
    nil
  end
```

- No need to use a local `result` variable
- Invert the `File.file?(key_file)` condition to reduce nested indentation
- `File.foreach` is lazy, so a big file can be used